modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue

```diff
@@ -266,56 +261,53 @@ export default {
     async create() {
       await this.edit(null);
     },
-    async import(action) {
-      const importOperation = this.utilityOperations.menu
+    async handleUtilityOperation(action) {
+      const operation = this.utilityOperations.menu
         .find((op) => op.action === action);
 
-      if (!importOperation) {
+      if (!operation) {
         return;
       }
 
-      const confirmed = await apos.confirm({
-        heading: this.$t('apostrophe:importPieces', { type: this.moduleLabels.singular }),
-        description: this.$t('apostrophe:importPiecesDescription'),
-        affirmativeLabel: this.$t('apostrophe:import'),
-        localize: false,
-        body: {
-          component: 'AposFile',
-          props: {
-            ...this.importPiecesFile,
-            allowedExtensions: '.csv'
-          },
-          on: {
-            'upload-file': this.uploadImportFile,
-            update: this.updateImportFile
-          }
-        }
-      });
-
-      if (confirmed && this.selectedFile) {
-        try {
-          const formData = new FormData();
-          formData.append('file', this.selectedFile);
+      const {
+        modal, body, ...modalOptions
+      } = operation.modalOptions || {};
 
-          this.selectedFile = null;
+      if (modal) {
+        apos.modal.execute(modal, {
+          action: this.moduleOptions.action,
+          route: operation.route,
+          labels: this.moduleLabels,
+          messages: operation.messages,
+          ...modalOptions
+        });
+      } else {
+        const confirmed = await apos.confirm({
+          heading: this.$t(modalOptions.title, { type: this.moduleLabels.plural }),
+          description: this.$t(modalOptions.description),
+          affirmativeLabel: this.$t(modalOptions.confirmationButton),
+          localize: false,
+          ...body && { body }
+        });
 
-          await apos.http.post(`${this.moduleOptions.action}${importOperation.route}`, {
-            body: formData
-          });
-        } catch (error) {
-          apos.notify('Utility operation {{ operation }} failed.', {
-            interpolate: { operation: importOperation.action },
-            type: 'danger'
-          });
+        if (confirmed) {
+          try {
+            await apos.http.post(`${this.moduleOptions.action}${operation.route}`, {
```

Comment:
Standard convention elsewhere in apostrophe is to insert the "/" between the action and route name here.

In particular the developer should NOT write "route: '/archive'" to indicate a route called archive in the current module. They should just write "route: 'archive'."

Otherwise it is very ambiguous what is meant by the "/".

If we do this right it will be possible at any point to support arbitrary URLs as well because we can immediately tell by the leading slash that they do *not* want the action prepended to them.

Reply:
This is related to how I originally implemented batch operations. @ValJed I changed that to use the batch operation name and removed the `route` option completely.

Reply:
Ok no problem I change that.